library/proc_macro/src/lib.rs

```diff
@@ -918,32 +918,29 @@ impl !Send for Punct {}
 #[stable(feature = "proc_macro_lib2", since = "1.29.0")]
 impl !Sync for Punct {}
 
-/// Indicates whether a `Punct` token can join with the following token
-/// to form a multi-character operator.
+/// Indicates whether a `Punct` token can join with the following token to form a new token.
```

Comment:
>a new token

It's not a token in the proc macro API sense, it's a larger parser-level structure.
I'd keep the previous wording, the result is multi-character and it's indeed an operator in most cases.
(Same below.)